### PR TITLE
User Logout Functionality

### PR DESCRIPTION
**Pull request by GabrielDushime**
What does this PR do?
-------------------------
This PR introduces enhancements to the logout functionality in our application. It adds robust testing for the logout endpoint to ensure its reliability and security.

Describe your changes
-------------------------
1.Added comprehensive tests for the logout functionality to cover various scenarios, including successful logout, invalid token handling, and unauthorized logout attempts.

2.Enhanced error handling in the verifyToken middleware to provide specific error messages for different JWT verification errors, such as malformed tokens.

3.Modified the Swagger documentation to include details about the logout endpoint and its usage.

How should this be manually tested?
----------------------------------------
1.Run the test suite using npm test and ensure that all logout-related tests pass without errors.

2.Manually test the logout functionality using tools like Postman or curl:
-Test logging out with a valid token.
-Test logging out without providing a token.
-Test logging out with an invalid token.

3.Verify that the Swagger documentation reflects the changes made to the logout endpoint.

What are the relevant pivotal tracker stories?
-------------------------------------------------
[Pivotal Tracker Link](https://www.pivotaltracker.com/n/projects/2703465/stories/187570730)

Screenshots (if is appropriate)
N/A

!Additional notes
-------------------
Please review the changes thoroughly, especially the additions to the logout functionality and the updated error handling in the middleware.

### CODE REVIEW

**Review by mutsinziisaac — DISMISSED** [review]
👍🏾

**Review by vegetason — DISMISSED** [review]
Good work so far as I don't see any problem

**Review by Mugisha146 — DISMISSED** [review]

File: src/middlewares/auth.middleware.ts [hunk 1]
```diff
@@ -50,6 +60,30 @@ export const restrictTo = (...roles: any) => {
...
+
+  try {
+    const decoded = jwt.verify(token, process.env.JWT_SECRET || 'default_secret');
+    req.user = decoded;
+    next();
+  } catch (error) {
+    console.error("Invalid token:", error);
+    return res.status(401).json({
+      status: "error",
+      message: "Invalid token.",
```
Comment (furebo) [inline src/middlewares/auth.middleware.ts:82]: Thank you for your work but Please remove all the commented codes.

**Review by furebo — CHANGES_REQUESTED** [review]

File: src/middlewares/auth.middleware.ts (hunk 1 above)
Comment (GabrielDushime) [reply src/middlewares/auth.middleware.ts]: Removed

**Review by SHYAKA-Aime — APPROVED** [review]
GREAT WORK GABRIEL

**Review by furebo — APPROVED** [review]

**Review by Mugisha146 — APPROVED** [review]
Well done

**Pull request merged**